src/main/java/net/snowflake/client/jdbc/SnowflakeFileTransferAgent.java

```diff
@@ -1268,6 +1287,83 @@ private static JsonNode parseCommandInGS(SFStatement statement, String command)
     return result;
   }
 
+  /**
+   * This is API function to parse the File Transfer Metadatas from a supplied PUT call response.
+   *
+   * <p>NOTE: It only supports PUT on S3/AZURE/GCS
+   *
+   * @param jsonNode JSON doc returned by GS from PUT call
+   * @return The file transfer metadatas for to-be-transferred files.
+   * @throws SnowflakeSQLException if any error occurs
+   */
+  public static List<SnowflakeFileTransferMetadata> getFileTransferMetadatas(JsonNode jsonNode)
+      throws SnowflakeSQLException {
+    CommandType commandType = CommandType.UPLOAD;
+    if (!jsonNode.path("data").path("command").isMissingNode()) {
+      commandType = CommandType.valueOf(jsonNode.path("data").path("command").asText());
+    }
+
+    if (commandType != CommandType.UPLOAD) {
+      throw new SnowflakeSQLException(
+          new Exception(),
+          SqlState.INTERNAL_ERROR,
+          ErrorCode.INTERNAL_ERROR.getMessageCode(),
+          "This API only supports PUT command");
+    }
+
+    JsonNode locationsNode = jsonNode.path("data").path("src_locations");
+
+    assert locationsNode.isArray();
+
+    final String[] src_locations;
+    final List<RemoteStoreFileEncryptionMaterial> encryptionMaterial;
+    try {
+      src_locations = mapper.readValue(locationsNode.toString(), String[].class);
+      encryptionMaterial = getEncryptionMaterial(commandType, jsonNode);
+    } catch (Exception ex) {
+      throw new SnowflakeSQLException(
+          ex,
+          SqlState.INTERNAL_ERROR,
+          ErrorCode.INTERNAL_ERROR.getMessageCode(),
+          "Failed to parse the locations due to: " + ex.getMessage());
+    }
+
+    final Set<String> sourceFiles;
```

Comment:
Nit, this could be reduced to a ternary:
```
Final Set<String> sourceFiles = commandType == CommandType.UPLOAD ? expandFileNames(srcLocations) : new Hashset<>...
```